Install the man pages

lmoureaux opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
This PR modifies the Makefile and setup script to install the man pages. They are installed at `$BASE/share/man`, where `$BASE` is as follows:

* `$VIRTUAL_ENV` when installed within a virtualenv
* `/usr` when installed system-wide

Man pages in the virtualenv can be accessed by setting the correct `$MANPATH`.

[outdated]
*A side-effect of this change is that I figured out how to modify the default `setuptools` install location. All scripts and macros are now installed in `$BASE/bin`. This means that `virtualenv` can take care of updating the `$PATH` (system-wide installs still need a setup script).*
[/outdated]

Another side-effect is that I corrected the license information in the package metadata. It was saying "MIT" while the `LICENSE` file always contained the text of GPLv3 (there is no way to change back to MIT without contacting every single contributor).

The setup script and developer documentation will need to be updated to use the `$MANPATH`.

### Downsides

* The `man` pages produced when the package is not installed may not be completely accurate. This is a problem for `make rpm` since it doesn't work inside a `virtualenv`.
* [outdated]*Macros and scripts are present twice in the package: under `bin/` and under `gempython/...`. I couldn't figure out why.*[/outdated]
* CI is broken :cry:

As a result of these downsides and my inability to fix them shortly, this PR is provided more as a list of commits to be cherry-picked.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue(s) here. -->
* #138
* [outdated] *Comment in `setup.py`:*
   ```
   would prefer to use this for executables, if one can control the install location
   to be user defined rather than /usr/bin
   ```
   *The preferred method (native support from `setuptools`) is now used.* [/outdated]

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
* The `pip` and `rpm` packages were built, the former from a `virtualenv` and the latter on `gem904daq01`.
* The `pip` package was installed; scripts and man pages are correctly installed and have the correct permissions.
* It was checked that the `rpm` package contains all files.
* The `man` pages built in each environment were (quickly) checked.

Since I don't have access to a `rpm`-based machine and I don't want to install one, I couldn't check that the `rpm` package installs cleanly.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--- Template thanks to https://www.talater.com/open-source-templates/#/page/99 -->

jsturdy commented on pkg/installrpm.sh:
```diff
@@ -23,8 +23,8 @@ cat INSTALLED_FILES.backup|egrep -v 'gempython/__init__.py*' > INSTALLED_FILES
 %attr(0755,root,root) /opt/cmsgemos/bin/*.py
-%attr(0755,root,root) /usr/lib/python*/site-packages/gempython/scripts/*.py
-%attr(0755,root,root) /usr/lib/python*/site-packages/gempython/gemplotting/macros/*.py
+%attr(0755,root,root) /opt/cmsgemos/lib/python*/site-packages/gempython/scripts/*.py
```
This was also explicitly avoided in the initial setup, as in future versions this will be replaced with system site-packages location macros provided by `rpm` (currently only works for `cc7`)

lmoureaux replied:
> system site-packages location macros provided by `rpm`

Could you provide more details about this? A link to the `rpm` doc about this would be great

jsturdy commented on pkg/setup.cfg:
```diff
@@ -4,7 +4,7 @@ name = gempython_gemplotting
 author_email = cms-gem-online-sw@cern.ch
 summary = __summary__
-license = MIT
+license = GNU General Public License v3 (GPLv3)
```
License changes *must* be discussed

lmoureaux replied:
This is not a license change, this is reflecting the fact that the current license *is* GPLv3. If you want to contact all copyright holders, fine; but until you get answers the license remains GPLv3.

jsturdy commented on pkg/setup.cfg:
```diff
@@ -20,7 +20,8 @@ classifier =
     Topic :: Scientific
     Topic :: Utilities
-    License :: OSI Approved :: MIT
+    License :: OSI Approved :: GNU General Public License (GPL)
```
Same

jsturdy commented on pkg/setup.py:
```diff
@@ -32,6 +38,20 @@ def getreqs():
+    # Man files
+    man_sections = {}
+    for file in listdir(mandir):
+        print('Found man page: %s' % file)
```
Also, prefer consistent string formatting (avoid `%` formatting in favour of `.format()` formatting) for all current and future contributions, `CONTRIBUTING.md` will be updated to reflect this and a few other changes

bdorney replied:
For my own ignorance, why is `.format()` viewed as superior?

jsturdy replied:
I don't know about *superior*... it is the standard that was introduced to eventually replace `%` formatting, though I suspect that `%` formatting will stick around for quite some time...
Both have a few advantages, though I think `.format()` has more.
The primary thing here (in this comment) is maintaining consistency

lmoureaux commented:
> avoiding putting things in macros into `/opt/cmsgemos/bin`

May I ask what's the rationale for the difference between `scripts` and `macros` in this project?

lmoureaux commented:
> I would have to go back in time to figure out why a given license was (probably arbitrarily) chosen for this project 

Commit 3d58f5d2 by... @jsturdy. The commit message ("Initial commit") isn't helpful in telling how the license was chosen.

FYI:
* CMSSW uses [Apache 2.0](http://www.apache.org/licenses/LICENSE-2.0).
* XDAQ is under a 3-clause [BSD-style license](https://svnweb.cern.ch/trac/cmsos/browser/trunk/LICENSE)

jsturdy commented:
> May I ask what's the rationale for the difference between scripts and macros in this project?

Mainly, it is to isolate the key executable parts (analyzing scans, which should go in `/opt/cmsgemos/bin`) from random plotting and style macros (can still be on the `PATH` for ease of use, provided `PATH` is modified to include the `macros` directory)

Further down the road, much of this will be replaced with `console_scripts` and `entry_points`, but that is a minor priority

lmoureaux commented:
Removed the controversial move to a more standard behavior using standard tools; the `rpm` package will now install `man` pages in `/usr/share`. Updated the first post accordingly.

lmoureaux commented:
Comments have been addressed and I'm waiting for another review.
